libr/bin/format/pyc/pyc_magic.c

```diff
@@ -3,226 +3,502 @@
+	case 0x0a0d0db0: // 3504
+		return &(PycVer){ 3, 12, "a1", "(Merge LOAD_METHOD back into LOAD_ATTR)" };
+	case 0x0a0d0db1: // 3505
+		return &(PycVer){ 3, 12, "a1", "(Specialization/Cache for FOR_ITER)" };
+	case 0x0a0d0db2: // 3506
+		return &(PycVer){ 3, 12, "a1", "(Add BINARY_SLICE and STORE_SLICE instructions)" };
+	case 0x0a0d0db3: // 3507
+		return &(PycVer){ 3, 12, "a1", "(Set lineno of module's RESUME to 0)" };
+	case 0x0a0d0db4: // 3508
+		return &(PycVer){ 3, 12, "a1", "(Add CLEANUP_THROW)" };
+	case 0x0a0d0db5: // 3509
+		return &(PycVer){ 3, 12, "a1", "(Conditional jumps only jump forward)" };
+	case 0x0a0d0db6: // 3510
+		return &(PycVer){ 3, 12, "a2", "(FOR_ITER leaves iterator on the stack)" };
+	case 0x0a0d0db7: // 3511
+		return &(PycVer){ 3, 12, "a2", "(Add STOPITERATION_ERROR instruction)" };
+	case 0x0a0d0db8: // 3512
+		return &(PycVer){ 3, 12, "a2", "(Remove all unused consts from code objects)" };
+	case 0x0a0d0db9: // 3513
+		return &(PycVer){ 3, 12, "a4", "(Add CALL_INTRINSIC_1 instruction, removed STOPITERATION_ERROR, PRINT_EXPR, IMPORT_STAR)" };
+	case 0x0a0d0dba: // 3514
+		return &(PycVer){ 3, 12, "a4", "(Remove ASYNC_GEN_WRAP, LIST_TO_TUPLE, and UNARY_POSITIVE)" };
+	case 0x0a0d0dbb: // 3515
+		return &(PycVer){ 3, 12, "a5", "(Embed jump mask in COMPARE_OP oparg)" };
+	case 0x0a0d0dbc: // 3516
+		return &(PycVer){ 3, 12, "a5", "(Add COMPARE_AND_BRANCH instruction)" };
+	case 0x0a0d0dbd: // 3517
+		return &(PycVer){ 3, 12, "a5", "(Change YIELD_VALUE oparg to exception block depth)" };
+	case 0x0a0d0dbe: // 3518
+		return &(PycVer){ 3, 12, "a6", "(Add RETURN_CONST instruction)" };
+	case 0x0a0d0dbf: // 3519
+		return &(PycVer){ 3, 12, "a6", "(Modify SEND instruction)" };
+	case 0x0a0d0dc0: // 3520
+		return &(PycVer){ 3, 12, "a6", "(Remove PREP_RERAISE_STAR, add CALL_INTRINSIC_2)" };
+	case 0x0a0d0dc1: // 3521
+		return &(PycVer){ 3, 12, "a7", "(Shrink the LOAD_GLOBAL caches)" };
+	case 0x0a0d0dc2: // 3522
+		return &(PycVer){ 3, 12, "a7", "(Removed JUMP_IF_FALSE_OR_POP/JUMP_IF_TRUE_OR_POP)" };
+	case 0x0a0d0dc3: // 3523
+		return &(PycVer){ 3, 12, "a7", "(Convert COMPARE_AND_BRANCH back to COMPARE_OP)" };
+	case 0x0a0d0dc4: // 3524
+		return &(PycVer){ 3, 12, "a7", "(Shrink the BINARY_SUBSCR caches)" };
+	case 0x0a0d0dc5: // 3525
+		return &(PycVer){ 3, 12, "b1", "(Shrink the CALL caches)" };
+	case 0x0a0d0dc6: // 3526
+		return &(PycVer){ 3, 12, "b1", "(Add instrumentation support)" };
+	case 0x0a0d0dc7: // 3527
+		return &(PycVer){ 3, 12, "b1", "(Add LOAD_SUPER_ATTR)" };
+	case 0x0a0d0dc8: // 3528
+		return &(PycVer){ 3, 12, "b1", "(Add LOAD_SUPER_ATTR_METHOD specialization)" };
+	case 0x0a0d0dc9: // 3529
+		return &(PycVer){ 3, 12, "b1", "(Inline list/dict/set comprehensions)" };
+	case 0x0a0d0dca: // 3530
+		return &(PycVer){ 3, 12, "b1", "(Shrink the LOAD_SUPER_ATTR caches)" };
+	case 0x0a0d0dcb: // 3531
+		return &(PycVer){ 3, 12, "b1", "(Add PEP 695 changes)" };
```

Comment:
Uhm this is returning a pointer in the stack. Thats not correct. What about returning the struct and use -1 in the first field instead of null on error?

Reply:
yeah... I was thinking backwards. I will update it. Thanks.